### PR TITLE
support for docker 1.3.2

### PR DESCRIPTION
**Pull request by andreaturli**
- update value objects (Container, Config)
- add new value object (Resource, StatusCode)
- add more options for ContainerApi
- add support for pause/unpause, restart, kill, copy and attach APIs + Mock and Live tests

### CODE REVIEW

File: docker/src/main/java/org/jclouds/docker/config/DockerOkHttpClientSupplier.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class DockerOkHttpClientSupplier implements OkHttpClientSupplier {
+
+   private final SSLContextWithKeysSupplier sslContextWithKeysSupplier;
+
+   @Inject
+   private DockerOkHttpClientSupplier(SSLContextWithKeysSupplier sslContextWithKeysSupplier) {
```
Comment (nacx) [inline docker/src/main/java/org/jclouds/docker/config/DockerOkHttpClientSupplier.java]: Make it package private by removint the `private` modifier:

```java
@Inject
DockerOkHttpClientSupplier(SSLContextWithKeysSupplier sslContextWithKeysSupplier) {
```

File: docker/src/main/java/org/jclouds/docker/suppliers/SSLContextWithKeysSupplier.java [hunk 2]
```diff
@@ -18,60 +18,154 @@
...
+   private static X509Certificate getCertificate(String certificate) {
+      try {
+         return (X509Certificate) CertificateFactory.getInstance("X.509").generateCertificate(
+                 new ByteArrayInputStream(certificate.getBytes(Charsets.UTF_8)));
+      } catch (CertificateException ex) {
+         throw new RuntimeException("Invalid certificate", ex);
+      }
+   }
+
+   private static PrivateKey getKey(String privateKey, String... password) {
```
Comment (nacx) [inline docker/src/main/java/org/jclouds/docker/suppliers/SSLContextWithKeysSupplier.java]: Hmmm still an array. Does it really need to be an array?

**Comment by nacx** [comment]
There just a couple remaining minors. lgtm once they're fixed. Thanks @andreaturli !

**Comment by andreaturli** [comment]
@nacx I think I need to rebase and squash so that you can merge it, yes? 
Thanks!

**Comment by andreaturli** [comment]
@nacx I think you are right, better to remove the password as we don't suggest to use encrypted PEM file

**Comment by nacx** [comment]
The build fails due to checkstyle violations. Address them while squashing and +1!

**Comment by andreaturli** [comment]
argh! @nacx 
Fixing them now

**Comment by andreaturli** [comment]
pushed to master as 74cefc97d0d62a921bcb1635be45153037d4c233

**Pull request closed without merging**